.github/workflows/main.yml

```diff
@@ -39,9 +39,28 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-node-
 
-      - name: Test
+      - name: Run unit tests
         run: ./gradlew check jibDockerBuild -x bootJar
 
+      - name: Run end-to-end tests
+        uses: cypress-io/github-action@v2
+        with:
+          working-directory: client
+          install: false
+          start: docker run -d -p8080:8080 --name=cfreak-cypress cfreak/codefreak
```

Comment:
We also have to mount the Docker socket and adjust the IP address of the reverse proxy. See #834.
We might be able to use the IP address of the default Docker network's gateway (= IP of the Host)
`docker network inspect bridge -f "{{ (index .IPAM.Config 0).Gateway }}"`